src/api/Controllers/CoursesController.cs

```diff
@@ -114,6 +115,19 @@ public IActionResult Index([FromBody]IList<Course> courses)
             if (ModelState.IsValid &&
                 courses.All(x => x.IsValid(false)))
             {
+                // circuit breaker for wildly different number of courses
+                var current = _context.Courses.Count();
+                if (current != 0) // empty database so skip the sanity check. This is useful for populating a local test system, or reconstructing production after a fire
+                {
+                    var incoming = courses.Count;
+                    var difference = incoming - current; // e.g. 100 existing, 98 incoming = difference of -2, i.e. there will be two less courses on find when completed
+                    const int maxDifference = 500; // todo: get from config, tune to realistic numbers. think about potential spikes at rollover etc
+                    if (Math.Abs(difference) > maxDifference)
+                    {
+                        _logger.LogError($"Refusing to update course list, difference of {difference} exceeded circuit breaker max of {maxDifference}");
```

Comment:
can we have some vital statistics before bad request

ie total current course count, new total course count, difference, and percentage of difference

Reply:
log informational output at start of every request with stats so we can monitor how close to threshold we are in normal operation